internal/venice-test-common/src/main/java/com/linkedin/venice/integration/utils/VeniceControllerCreateOptions.java

```diff
@@ -0,0 +1,291 @@
+import static com.linkedin.venice.integration.utils.VeniceClusterWrapperConstants.DEFAULT_PARTITION_SIZE_BYTES;
+import static com.linkedin.venice.integration.utils.VeniceClusterWrapperConstants.DEFAULT_REPLICATION_FACTOR;
+import static com.linkedin.venice.integration.utils.VeniceControllerWrapper.*;
+
+import com.linkedin.venice.authorization.AuthorizerService;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import org.apache.commons.lang.Validate;
+
+
+public class VeniceControllerCreateOptions {
+  private final boolean isParent;
+  private final boolean sslToKafka;
+  private final boolean d2Enabled;
+  private final int replicationFactor;
+  private final int partitionSize;
+  private final int minActiveReplica;
+  private final long rebalanceDelayMs;
+  private final String[] clusterNames;
+  private final String zkAddress;
+  private final String clusterToD2;
+  private final VeniceControllerWrapper[] childControllers;
+  private final KafkaBrokerWrapper kafkaBroker;
+  private final Properties extraProperties;
+  private final AuthorizerService authorizerService;
+
+  private VeniceControllerCreateOptions(Builder builder) {
+    isParent = builder.isParent;
+    sslToKafka = builder.sslToKafka;
+    d2Enabled = builder.d2Enabled;
+    replicationFactor = builder.replicationFactor;
+    partitionSize = builder.partitionSize;
+    minActiveReplica = builder.minActiveReplica;
+    rebalanceDelayMs = builder.rebalanceDelayMs;
+    clusterNames = builder.clusterNames;
+    zkAddress = builder.zkAddress;
+    clusterToD2 = builder.clusterToD2;
+    childControllers = builder.childControllers;
+    kafkaBroker = builder.kafkaBroker;
+    extraProperties = builder.extraProperties;
+    authorizerService = builder.authorizerService;
+  }
+
+  @Override
+  public String toString() {
+    return new StringBuilder().append("isParent:")
+        .append(isParent)
+        .append(", ")
+        .append("sslToKafka:")
+        .append(sslToKafka)
+        .append(", ")
+        .append("replicationFactor:")
+        .append(replicationFactor)
+        .append(", ")
+        .append("partitionSize:")
+        .append(partitionSize)
+        .append(", ")
+        .append("minActiveReplica:")
+        .append(minActiveReplica)
+        .append(", ")
+        .append("rebalanceDelayMs:")
+        .append(rebalanceDelayMs)
+        .append(", ")
+        .append("clusterNames:")
+        .append(Arrays.toString(clusterNames))
+        .append(", ")
+        .append("zkAddress:")
+        .append(zkAddress)
+        .append(", ")
+        .append("kafkaBroker:")
+        .append(kafkaBroker == null ? "null" : kafkaBroker.getAddress())
+        .append(", ")
+        .append("d2Enabled:")
+        .append(d2Enabled)
+        .append(", ")
+        .append("clusterToD2:")
+        .append(clusterToD2)
+        .append(", ")
+        .append("extraProperties:")
```

Comment:
nit: is `extraProperties` human-readable?

Reply:
Yes. Here `extraProperties` is a variable of type `Properties` which extends `Hashable`. And `Hashable` provides toString() implementation.